server/src/main/java/io/littlehorse/server/streams/storeinternals/ReadOnlyGetableManager.java

```diff
@@ -84,6 +85,23 @@ protected <U extends Message, T extends CoreGetable<U>> List<GetableToStore<U, T
         return store.range(start, end, cls);
     }
 
+    /**
+     * Iterate over tags
+     */
+    public LHKeyValueIterator<Tag> tagScan(String start, String end) {
+        return store.range(start, end, Tag.class);
+    }
+
+    /**
+     * Only use this method if you need to retrieve a Storeable
+     * @param storeKey Storeable's storekey
+     * @param cls Storeable's class
+     * @return Null if not found
+     */
+    public <U extends Message, T extends Storeable<U>> T get(String storeKey, Class<T> cls) {
```

Comment:
@coltmcnealy-lh  This is leaking the implementation detail. Any suggestion here? see `OneTaskQueue#rehydrateFromStore`

Reply:
Can we just rename it to `getStoreable()`? I don't like overloading the `get()` method, which is used to get a `Getable`.

Reply:
Actually, try this:

```
public ScheduledTaskModel getScheduledTask(TaskRunIdModel scheduledTaskId);
```

That still leaks, but it leaks a lot less.

Reply:
That leaks the business application into the store abstraction. ;)